### ISSUE TITLE
With Java 8 update 171 (32 bits) NVDA reports "Java access bridge is not available" in the log viewer and fails to read any Java apps

### ISSUE THREAD

**Issue by Palacee-hun** [nvaccess/nvda#8454]
### Steps to reproduce:
Clearly stated in the title. Java Access Bridge was also manually enabled by running "jabswitch /enable"and this didn't help.
### Actual behavior:
Clearly stated in the title.
### Expected behavior:
Java access should work normally as it ever did. Java 8 update 151 (the version I last had before the latest Java update) worked without problems.
#### NVDA Installed/portable/running from source:
installed
#### NVDA version:
2018.2.1
#### Windows version:
Windows 7 Ultimate 32-bits Hungarian
#### Name and version of other software in use when reproducing the issue:
Java 8 update 171 32-bits
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA?
Yes, NVDA 2018.1.1 produces the exact same.

**Comment by Palacee-hun** [comment]
Here is my NVDA log with NVDA start-up messages, plus the developer info of the Java Control Panel window.
[nvda.log](https://github.com/nvaccess/nvda/files/2141598/nvda.log)

**Comment by leonardder** [comment]
I've seen several cases where a JAVA update broke the access bridge, 
probably because it was not possible for the update installer to replace 
an essential dll. Please perform a removal of JAVA, then restart your 
system, make sure that c:\windows\syswow64 doesn't contain any files 
starting with WindowsAccessBridge and try to install the product again.

**Comment by Palacee-hun** [comment]
Thanks much, I followed your suggestion and yes, Java access is working again as expected.

For documentation purposes:

After removing Java and rebooting I found two dll files starting with windowsaccessbridge (and then a seven-digit number) in my c:\windows\system32 directory (as I have stated I've got a 32-bit copy of Windows). I moved these to a non-system backup directory and then reinstalled Java from the official website. Then restarted NVDA and Java access got restored. Running jabswitch wasn't needed.

So the Java updater seems a no-go for screen reader users. Sad news. This is quite a time consuming and cumbersome workaround.

I leave this issue open for two reasons. Firstly this problem and workaround should be documented officially in NVDA due to its severity. After all it's much more than a simple nuisance to have Java accessibility broken by an update procedure which Oracle strongly endorses, almost requires from time to time. Secondly Java 7 updates never did this. So maybe NVDA devs could learn something from this and might adapt to this in NVDA somehow if possible, perhaps through a different handling and releasing of the Java access bridge dll.

**Comment by Brian1Gaff** [comment]
I think I've read about similar issues with java for the sighted with other 
bits of code not working, on a forum somewhere quite recently. Sounds like 
java should not make updates they should make a removal tool!
I gave up on Java. if I find I really need it, I'll put it back otherwise. 
no, life is too short!
 Brian

**Comment by leonardder** [comment]
Closing as works for me
